src/textual/css/_style_properties.py

```diff
@@ -115,12 +115,12 @@ def __set__(
         ):
             raise StyleValueError("'auto' not allowed here")
 
-        if new_value.unit != Unit.AUTO:
-            if new_value is not None and new_value.unit not in self.units:
+        if new_value is not None and new_value.unit != Unit.AUTO:
```

Comment:
PyCharm was highlighting the fact that `if new_value.unit != Unit.AUTO`  will raise an exception if "new_value" is `None` anyway, so I guess we'd better wrap the whole block under a "is not None" guard?